integration-cli/requirement/requirement.go

```diff
@@ -0,0 +1,19 @@
+package requirement
+
+type skipT interface {
+	Skip(reason string)
+}
+
+// Test represent a function that can be used as a requirement validation.
+type Test func() (bool, string)
+
+// Is checks if the environment satisfies the requirements
+// for the test to run or skips the tests.
+func Is(s skipT, requirements ...Test) {
+	for _, r := range requirements {
+		isValid, message := r()
+		if !isValid {
+			s.Skip(message)
```

Comment:
If that's the only place where the message is used, then _maybe_ we should consider a nuclear option that would avoid complicating the syntax for all the callers of requirement functions. We could:

- Have requirements functions be `func () bool`, in such way that we can continue writing `if SameHostDaemon() {`.
- Use some reflect magic here to provide a message, as in `s.Skip(fmt.Sprintf("unmatched requirement %s", runtime.FuncForPC(<some_reflect_magic>).Name())`.